pkg/templates/nodelocaldns/nodelocaldns.go

```diff
@@ -187,6 +187,10 @@ func dnscacheDaemonSet() *appsv1.DaemonSet {
 	trueBool := true
 	hostPathFileOrCreate := corev1.HostPathFileOrCreate
 	terminationGracePeriodSeconds := int64(0)
+	execScript := fmt.Sprintf(`
+sleep 10;
+exec /node-cache -localip %s -conf /etc/Corefile -upstreamsvc kube-dns-upstream`,
+		VirtualIP)
```

Comment:
Would it make sense that instead of sleeping 10 seconds, we poll for the kube-proxy DaemonSet/pods before deploying NodeLocalDNS and wait for all replicas to be ready? I think that would be a safer approach, especially if kube-proxy needs more than 10 seconds to come up for any reason.

If we decide to stick with this approach, can we add a comment describing why we did this?

Reply:
No, because the racecondition also kicks in after node reboot, when there is no kubeone involved.

Reply:
Okay, makes sense. Please add a comment describing that above this line.

Reply:
added an explanatory comment